portal/server/routes/default/index.js

```diff
@@ -6,13 +6,20 @@ const router = express.Router();
 
 router.get('/', (req, res) => {
   const { user } = req.session;
-  if (!user) {
-    return res.redirect('/login');
+
+  /**
+   * Need to ensure `roles` object exist.
+   * When a user is first logged-in as a TFM user,
+   * roles object will not exist and will throw 500.
+   */
```

Comment:
I'm finding this confusing - how/why, if a user is logged in - should we redirect to `/login`?

Reply:
This is because if a user is logged-in locally as a TFM user and then in the same session login as a Portal user `roles` object will not exists, this mitigate that specific edge cases and asks users to login into Portal.

Reply:
Got it, make sense, thanks - didn't realise this was in Portal.